src/go/intelligence/pipeline/fetch_alert.go

```diff
@@ -21,7 +15,12 @@ import (
 func FetchAlert() <-chan models.AmJSON {
 	fetchedData := make(chan models.AmJSON)
 	go func() {
```

Comment:
the same problem with fetchedData channel, it may never be closed if the body of this function fails. Please use `defer close(fetchedData)` at the beginning of this goroutine.